src/components/media/controls/controls.tsx

```diff
@@ -0,0 +1,274 @@
+        data-cy="volume-slider"
+        size="small"
+        max={1}
+        step={0.1}
+        aria-label="Volume"
+        aria-valuetext={`Volume: ${volumeFraction * percentMultiple}%`}
+        value={volumeFraction}
+        onChange={handleVolumeChange}
+      />
+    </Box>
+  )
+}
+
+interface TranscriptToggleProps {
+  isTranscriptShown: boolean
+  setIsTranscriptShown: () => void
+}
+
+export function TranscriptToggle(props: TranscriptToggleProps) {
+  const Icon = props.isTranscriptShown
+    ? KeyboardArrowUpIcon
+    : KeyboardArrowDownIcon
+
+  const buttonText = `${props.isTranscriptShown ? 'Hide' : 'Show'} Transcript`
+
+  return (
+    <Button
+      className="rustic-transcript-toggle"
+      data-cy="transcript-toggle"
+      onClick={props.setIsTranscriptShown}
+      endIcon={<Icon />}
+    >
+      <Typography variant="overline">{buttonText}</Typography>
+    </Button>
+  )
+}
+
+export function PausePlayToggle(props: MediaControls) {
+  const [isPlaying, setIsPlaying] = useState(false)
+
+  const action = isPlaying ? 'pause' : 'play'
+
+  function handlePausePlayToggle() {
+    if (isPlaying) {
+      props.mediaElement.pause()
+      setIsPlaying(false)
+    } else {
+      props.mediaElement.play()
+      setIsPlaying(true)
+    }
+  }
+
+  props.mediaElement.onended = function () {
+    setIsPlaying(false)
+  }
+
+  return (
+    <MediaIconButton
+      onClick={handlePausePlayToggle}
+      action={action}
+      className="rustic-pause-play-icon"
+    />
+  )
+}
+
+export function PlaybackRateButton(props: MediaControls) {
+  const [playbackRate, setPlaybackRate] = useState(1)
+
+  function handlePlaybackRateChange() {
+    let newPlaybackRate = 1
+    const rateChange = 0.5
+    const maxRate = 2
+    // Increase playback speed by 0.5 until until 2 is reached, then reset to 1
+    if (props.mediaElement.playbackRate < maxRate) {
+      newPlaybackRate = props.mediaElement.playbackRate + rateChange
+    }
+
+    props.mediaElement.playbackRate = newPlaybackRate
+    setPlaybackRate(newPlaybackRate)
+  }
+
+  return (
+    <Button
+      onClick={handlePlaybackRateChange}
+      className="rustic-playback-rate-button"
+      aria-label={`Playback rate: ${playbackRate}x, click to change`}
+      data-cy="playback-rate-button"
+    >
+      <Typography variant="body1" color="primary.main">
+        {playbackRate}X
+      </Typography>
+    </Button>
+  )
+}
+
+interface MoveTenSecondsButtonProps extends MediaControls {
+  movement: 'replay' | 'forward'
+}
+
+export function MoveTenSecondsButton(props: MoveTenSecondsButtonProps) {
+  const isReplayMovement = props.movement === 'replay'
+
+  const onClick = isReplayMovement
+    ? handleReplayTenSeconds
+    : handleForwardTenSeconds
+
+  const action = isReplayMovement ? 'replay' : 'forward'
+
+  function handleForwardTenSeconds() {
+    props.mediaElement.currentTime += 10
+  }
+
+  function handleReplayTenSeconds() {
+    props.mediaElement.currentTime -= 10
+  }
```

Comment:
Could you move these two functions to the top of `onClick`? It's nice to define the functions first before using them